internal/pkg/core/statsmanager.go

```diff
@@ -289,25 +289,106 @@ func (StatMgr *OpenOltStatisticsMgr) collectNNIMetrics(nniID uint32) map[string]
 
 	if metricName != nil && len(metricName) > 0 {
 		for mName := range metricName {
-			switch mName {
-			case "rx_bytes":
-				nnival["RxBytes"] = float32(cm.RxBytes)
-			case "rx_packets":
-				nnival["RxPackets"] = float32(cm.RxPackets)
-			case "rx_ucast_packets":
-				nnival["RxUcastPackets"] = float32(cm.RxUcastPackets)
-			case "rx_mcast_packets":
-				nnival["RxMcastPackets"] = float32(cm.RxMcastPackets)
-			case "rx_bcast_packets":
-				nnival["RxBcastPackets"] = float32(cm.RxBcastPackets)
-			case "tx_bytes":
-				nnival["TxBytes"] = float32(cm.TxBytes)
-			case "tx_packets":
-				nnival["TxPackets"] = float32(cm.TxPackets)
-			case "tx_mcast_packets":
-				nnival["TxMcastPackets"] = float32(cm.TxMcastPackets)
+                        switch mName {
+                        case "rx_bytes":
+                                if metricName["rx_bytes"].Enabled == true {
+                                        nnival["RxBytes"] = float32(cm.RxBytes)
+                                } else {
+                                        _, ok := nnival["RxBytes"]
+                                        if ok {
+                                                delete(nnival, "RxBytes")
+                                                }
+                                        }
+                        case "rx_packets":
+                                if metricName["rx_packets"].Enabled == true {
+                                        nnival["RxPackets"] = float32(cm.RxPackets)
+                                } else {
+                                        _, ok := nnival["RxPackets"]
+                                        if ok {
+                                                delete(nnival, "RxPackets")
+                                                }
+                                        }
+                        // these are not supported in OpenOlt Agent now
+                        // will return zero until supported
+                        case "rx_ucast_packets":
+                                if metricName["rx_ucast_packets"].Enabled == true {
+                                        nnival["RxUcastPackets"] = float32(cm.RxUcastPackets)
+                                } else {
+                                        _, ok := nnival["RxUcastPackets"]
+                                        if ok {
```

Comment:
it can be wriitten else if _,ok := nnival["RxUcastPackets"];ok{
        delete()
}
follow the same below